catalog/transformer/catalog-transformer-propertyjson/src/main/java/ddf/catalog/transformer/input/propertyjson/PropertyJsonInputTransformer.java

```diff
@@ -0,0 +1,242 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.transformer.input.propertyjson;
+
+import ddf.catalog.data.AttributeDescriptor;
+import ddf.catalog.data.AttributeInjector;
+import ddf.catalog.data.AttributeType;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.MetacardType;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.data.impl.MetacardImpl;
+import ddf.catalog.data.types.Core;
+import ddf.catalog.transform.CatalogTransformerException;
+import ddf.catalog.transform.InputTransformer;
+import java.io.InputStream;
+import java.io.Serializable;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.TimeZone;
+import org.boon.json.JsonFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PropertyJsonInputTransformer implements InputTransformer {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(PropertyJsonInputTransformer.class);
+
+  private static final String ISO_8601_DATE_FORMAT = "yyyy-MM-dd'T'HH:mm:ss.SSSZ";
+
+  private final List<MetacardType> metacardTypeList = new LinkedList<>();
+
+  private final MetacardType defaultMetacardType;
+
+  private final AttributeInjector attributeInjector;
+
+  public PropertyJsonInputTransformer(
+      MetacardType defaultMetacardType, AttributeInjector attributeInjector) {
+
+    this.attributeInjector = attributeInjector;
+    this.defaultMetacardType = attributeInjector.injectAttributes(defaultMetacardType);
+  }
+
+  public void addMetacardType(MetacardType metacardType) {
+    this.metacardTypeList.add(attributeInjector.injectAttributes(metacardType));
+  }
+
+  public void removeMetacardType(MetacardType metacardType) {
+    this.metacardTypeList.removeIf(mt -> metacardType.getName().equals(mt.getName()));
+  }
+
+  @Override
+  public Metacard transform(InputStream input) throws CatalogTransformerException {
+    return transform(input, null);
+  }
+
+  @SuppressWarnings("unchecked")
+  @Override
+  public Metacard transform(InputStream input, String id) throws CatalogTransformerException {
```

Comment:
❗️ There is a lot of duplication between this transformer and GeoJSON.  I think we can support this use case with the GeoJSON transformer.  We would be able to remove the propertyjson transformer code if that works.

>GeoJSON processors MAY interpret Geometry objects with empty "coordinates" arrays as null objects.
https://tools.ietf.org/html/rfc7946#section-3.1

Reply:
done